lib/bolt/inventory/group2.rb

```diff
@@ -355,81 +363,92 @@ def validate(used_names = Set.new, target_names = Set.new, aliased = {}, depth =
         nil
       end
 
-      # The data functions below expect and return nil or a hash of the schema
-      # {'config' => Hash, 'vars' => Hash, 'facts' => Hash, 'features' => Array,
-      #  'plugin_hooks' => Hash, 'groups' => Array}
-      def data_for(target_name)
-        data_merge(group_collect(target_name), target_collect(target_name))
+      def resolve_data_keys(data, target = nil)
+        result = {
+          'config' => resolve_references(data.fetch('config', {})),
```

Comment:
Is this functionally different from `result = resolve_references(data)` at or `empty_data.merge(data)` this point?

Reply:
We pass this `@input` on the group, which is the entire initial hash including the _unresolved_ data from `targets` and `groups`. If we just did `resolve_references(@data)`, that would resolve the targets and groups again.

It could make sense to do something like `@unresolved_data = input.select { ... }` with just the `DATA_KEYS`, instead of having `@input`. That doesn't extend well to targets though, since we don't have an object to store data separate from name/uri/alias...